content/blog/js-beginners-guide.md

```diff
@@ -0,0 +1,107 @@
+---
+author: "Alex Hill"
+date: 2020-02-02
+title: Front-end Javascript Frameworks - A Beginner's Guide
+tags:
+ - javascript
+ - vue
+ - react
+ - angular
+---
+
+We've put together a little resource for RSEs (or anyone else!) who are writing browser 
+based apps and want to be using the latest Javascript web technologies, but aren't sure 
+where to start. 
+
+The full <a href="/resources/jsbeginnersguide.pdf" target="_blank">poster</a> was presented 
+at [RSLondonSouthEast](https://rslondon.ac.uk/) on 6th February.
+
+## Key concepts 
+If you've just gotten started with the world of front-end frameworks, you've probably heard of 
+**components** and **directives**. These are the key building blocks for modern JS frameworks.
+
+A component is like a custom HTML element with special behaviour attached:
+
+```
+<error-list errors="myErrorArray"></error-list>
+```
+
+A directive is like a custom data attribute on a regular HTML element, with special behaviour attached:
+
+```
+<span v-if="hasError" v-text="message"></span>
+```
+
+A component is a combination of a **template** that defines the presentation, and a script that defines 
+behaviour.
```

Comment:
"behaviour and data"? Could include a really simple script which corresponds to your error list template to show what that might look like?